crates/bindings-macro/src/view.rs

```diff
@@ -0,0 +1,284 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::parse::Parser;
+use syn::{FnArg, ItemFn};
+
+use crate::sym;
+use crate::util::{ident_to_litstr, match_meta};
+
+pub(crate) struct ViewArgs {
+    anonymous: bool,
+}
+
+impl ViewArgs {
+    /// Parse `#[view(public)]` where public is required
+    pub(crate) fn parse(input: TokenStream) -> syn::Result<Self> {
+        if input.is_empty() {
+            return Err(syn::Error::new(
+                proc_macro2::Span::call_site(),
+                "views must be declared as #[view(public)]; public is required",
+            ));
+        }
```

Comment:
I would look at `TableArgs::parse`. I think you should collect an `access: Option<TableAccess>` and e.g., `Option<Span>`  for `anonymous` and then ensure that at the end, `access.is_some()` or error.